selftests/functional/test_basic.py

```diff
@@ -0,0 +1,123 @@
+            - pass:
+                result_param = 'pass'
+            - fail:
+                result_param = 'fail'
+            - error:
+                result_param = 'error'
+            - autotest_skip:
+                result_param = 'autotest_skip'
+            - autotest_pass:
+                result_param = 'autotest_pass'
+            - autotest_fail:
+                result_param = 'autotest_fail'
+            - autotest_error:
+                result_param = 'autotest_error'
+            - other_exception:
+                result_param = "other exception"
+
+only test_statuses
+kvm_ver_cmd = /bin/true
+kvm_userspace_ver_cmd = /bin/true
+verify_host_dmesg = no
+"""
+
+
+class BasicTests(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+        self.rm_files = []
+
+    def test_statuses(self):
+        os.chdir(BASE_DIR)
+        test_path = os.path.join(data_dir.get_test_providers_dir(),
+                                 "downloads", "io-github-autotest-qemu",
+                                 "generic", "tests", "test_statuses.py")
+        self.assertTrue(os.path.exists(os.path.dirname(test_path)),
+                        "The qemu providers dir does not exists, Avocado-vt "
+                        "is probably not configured properly.")
+        self.rm_files.append(test_path)
+        test = script.make_script(test_path, TEST_STATUSES_PY)
+        cfg = script.make_script(os.path.join(self.tmpdir,
+                                              "test_statuses.cfg"),
+                                 TEST_STATUSES_CFG)
+        result = process.run("avocado --show all run --vt-config %s "
+                             "--job-results-dir %s"
+                             % (cfg, self.tmpdir), ignore_status=True)
+        self.assertEqual(result.exit_status, 1, "Exit status is not 1:\n%s"
+                         % result)
+        status = json.load(open(os.path.join(self.tmpdir, "latest",
+                                             "results.json")))
+        act_statuses = [_["status"] for _ in status["tests"]]
+        statuses_master = ["SKIP", "PASS", "FAIL", "ERROR", "CANCEL", "PASS",
+                           "FAIL", "ERROR", "ERROR"]
+        statuses_36lts = ["SKIP", "PASS", "FAIL", "ERROR", "SKIP", "PASS",
+                          "FAIL", "ERROR", "ERROR"]
+        if not (act_statuses == statuses_master or
+                act_statuses == statuses_36lts):
+            self.assertTrue(False, "Test statuses does not match any of "
+                            "expected results:\nmaster: %s\n36lts: %s\n"
+                            "actual: %s\n\noutput:\n%s"
+                            % (statuses_master, statuses_36lts, act_statuses,
+                               result))
+
+    def tearDown(self):
+        for path in self.rm_files:
+            try:
+                os.unlink(path)
+            except IOError:
+                pass
+
```

Comment:
I see, but there are both the `rm_files` and the `tmpdir`.  My suggestion is about the second one, to pair with it with the `tmpdir` creation.